File: 1028-interval-list-intersections/1028-interval-list-intersections.java
class Solution {
    public int[][] intervalIntersection(int[][] firstList, int[][] secondList) {

        int left = 0;
        int right = 0;
        List<int[]> list = new ArrayList<>();

        while(left <firstList.length &&  right < secondList.length){
            int start = Math.max(firstList[left][0], secondList[right][0]);
            int end = Math.min(firstList[left][1], secondList[right][1]);

            if(start <= end){
                list.add(new int[] {start, end});
            }

            if(firstList[left][1] < secondList[right][1]){
                left++;
            }else{
                right++;
            }


        }

        return list.toArray(new int[list.size()][]);
        
    }
} 

File: 1028-interval-list-intersections/1028-interval-list-intersections.js
/**
 * @param {number[][]} firstList
 * @param {number[][]} secondList
 * @return {number[][]}
 */
var intervalIntersection = function(firstList, secondList) {

    let i = 0;
    let j = 0;

    let res = [];

    while(i < firstList.length && j< secondList.length){
        let start = Math.max(firstList[i][0], secondList[j][0]);
        let end =  Math.min(firstList[i][1], secondList[j][1]);

        if(start <= end ){
            res.push([start, end]);
        }

        if(firstList[i][1] < secondList[j][1]){
            i++;
        }else{
            j++;
        }
    }

    return res;
    
};